docs/explanation/dump_plugin.rst

```diff
@@ -0,0 +1,42 @@
+.. _dump_plugin_explanation:
+
+Dump Plugin
+===========
+
+The dump plugin is a simple plugin that by default is equivalent to run the
+following command from the source directory:
+
+.. code-block:: shell
+    
+    cp --archive --link --no-dereference . "${CRAFT_PART_INSTALL}"
+
+
+The :ref:`source <source>` property is the key to this plugin. With some
+additional properties, they define the location where to get the files from, or
+which branch, tag, and/or commit to clone if it is a repository.
+
+
+If you are not using the source files directly in the final payload, but want to
+run some custom commands to generate them. Then you should use the ``nil``
+plugin instead to avoid copying any unnecessary files.
+
+
+Combining the dump plugin with the :ref:`part properties <part_properties>`
+allows extending the behavior to create new files, modify existing files,
+and/or filter files for the final payload.
+
+
+For example, it could be used with the :ref:`override-build <override_build>`,
+to convert file formats. 
+
+
+.. ifconfig:: project in ["Snapcraft"]
```

Comment:
I think we should move away from this kind of project-specific customization in craft-parts. Instead, I think these contents should be in `base/explanation/dump_plugin.rst` so that Snapcraft itself can customize it